### PR TITLE
Adds example to `mediapipe-task-text`

### PR DESCRIPTION
**Pull request by craiglabenz**
Adds:
 * `example` directory to `mediapipe-task-text`
 * `sdks_finder` command to find and declare from where apps should download SDKs
 * expanded CI logic to run `example` tests on GitHub actions (requires output from `sdks_finder` command)

### CODE REVIEW

File: packages/mediapipe-core/lib/src/task_options.dart [hunk 1]
```diff
@@ -21,51 +21,84 @@ import 'third_party/mediapipe/generated/mediapipe_common_bindings.dart'
...
   /// MediaPipe SDK to instantiate various classifiers.
   Pointer<bindings.BaseOptions> toStruct() {
     final struct = calloc<bindings.BaseOptions>();
 
-    if (modelAssetPath != null) {
+    if (_type == _BaseOptionsType.path) {
       struct.ref.model_asset_path = prepareString(modelAssetPath!);
     }
-    if (modelAssetBuffer != null) {
+    if (_type == _BaseOptionsType.memory) {
```
Comment (Piinks) [inline packages/mediapipe-core/lib/src/task_options.dart:75]: These should be a switch statement. That way we will know all cases are exhaustively covered, and the analyzer will flag areas that need an update if we ever add another _BaseOptionsType.

May need to turn on https://dart.dev/tools/linter-rules/exhaustive_cases in the analysis file if it is not already

File: packages/mediapipe-task-text/example/lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      // Replace "..." with the results
+      _log.info('"$_isProcessing" $categoryName :: $score');
+      results.last = Card(
+        child: Padding(
+            padding: const EdgeInsets.all(10),
+            child: Text('"$_isProcessing" $categoryName :: $score')),
+      );
+      _isProcessing = null;
+    });
+    setState(() {});
```
Comment (Piinks) [inline packages/mediapipe-task-text/example/lib/main.dart:76]: Empty setState calls are discouraged. What is actually changing here?

File: packages/mediapipe-task-text/example/lib/main.dart [hunk 3]
```diff
@@ -0,0 +1,109 @@
+import 'dart:io' as io;
```
Comment (Piinks) [inline packages/mediapipe-task-text/example/lib/main.dart:1]: This example needs tests!

**Review by Piinks — COMMENTED** [review]
What is packages/mediapipe-task-text/example/assets/bert_classifier.tflite1?

File: packages/mediapipe-task-text/example/lib/main.dart (hunk 2 above)
Comment (craiglabenz) [reply packages/mediapipe-task-text/example/lib/main.dart:76]: Ah yes, this goes back to an earlier commit message that read:

```
native assets troubleshooting
this commit is broken
```

I'll give this whole PR another pass in light of that commit and make sure there's no other cruft.

**Review by PaulTR — DISMISSED** [review]
Just requesting changes to add to the whole multiple-commits experience

**Review by domesticmouse — APPROVED** [review]

File: packages/mediapipe-core/lib/src/task_options.dart (hunk 1 above)
Comment (Piinks) [reply packages/mediapipe-core/lib/src/task_options.dart:75]: Just bumping this

File: packages/mediapipe-task-text/example/lib/main.dart (hunk 2 above)
Comment (Piinks) [reply packages/mediapipe-task-text/example/lib/main.dart:106]: Bump here :)

File: packages/mediapipe-task-text/test/text_classifier_result_test.dart [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import 'package:mediapipe_core/mediapipe_core.dart';
+import 'package:mediapipe_text/mediapipe_text.dart';
+import 'package:mediapipe_core/src/third_party/mediapipe/generated/mediapipe_common_bindings.dart'
+    as core_bindings;
+import 'package:mediapipe_text/src/third_party/mediapipe/generated/mediapipe_text_bindings.dart'
+    as bindings;
+
+void main() {
+  group('TextClassifierResult.structToDart should', () {
+    // test('load an empty object', () {
```
Comment (Piinks) [inline packages/mediapipe-task-text/test/text_classifier_result_test.dart:13]: Should this test be enabled?

File: packages/mediapipe-task-text/test/text_classifier_result_test.dart [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        buildClassifications(index: 2),
+      ];
+      struct.ref.classifications_count = classifications.length;
+      struct.ref.classifications = classifications.first;
+      struct.ref.has_timestamp_ms = true;
+      struct.ref.timestamp_ms = 9999999;
+
+      final result = TextClassifierResult.structToDart(struct.ref);
+      // expect(result.classifications, hasLength(2));
+      // expect(result.timestamp, isNotNull);
```
Comment (Piinks) [inline packages/mediapipe-task-text/test/text_classifier_result_test.dart:39]: How about these expectations?

File: tool/ci_script_shared.sh [hunk 6]
```diff
@@ -16,11 +32,56 @@ function ci_package () {
 
         # Run the formatter on all the dart files to make sure everything's linted.
         dart format --output none --set-exit-if-changed .
+        
+        flutter config --enable-native-assets
+
+        echo "$ flutter config --list 1"
```
Comment (Piinks) [inline tool/ci_script_shared.sh:38]: I remember we added oodles of echos for debugging CI, do we need them now?

File: tool/mediapipe_ci_script_beta.sh [hunk 7]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-set -e
+# set -e
```
Comment (Piinks) [inline tool/mediapipe_ci_script_beta.sh:3]: Remove or uncomment?

File: tool/mediapipe_ci_script_master.sh [hunk 8]
```diff
@@ -1,14 +1,15 @@
 #!/bin/bash
 
-set -e
+# set -e
```
Comment (Piinks) [inline tool/mediapipe_ci_script_master.sh:3]: Same here

File: tool/mediapipe_ci_script_master.sh [hunk 9]
```diff
@@ -1,14 +1,15 @@
...
 
 DIR="${BASH_SOURCE%/*}"
 source "$DIR/ci_script_shared.sh"
 
 flutter doctor -v
 
 declare -ar PACKAGE_NAMES=(
-    "mediapipe-core"
+    # TODO(craiglabenz): Uncomment once native assets works on CI
+    # "mediapipe-core"
```
Comment (Piinks) [inline tool/mediapipe_ci_script_master.sh:12]: Aha! Is this the current blocker?

File: tool/mediapipe_ci_script_stable.sh [hunk 10]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-set -e
+# set -e
```
Comment (Piinks) [inline tool/mediapipe_ci_script_stable.sh:3]: ++

Comment (craiglabenz) [reply tool/mediapipe_ci_script_stable.sh:3]: This change is temporary to increase debugging visibility into what's happening on GitHub Actions machines

File: tool/mediapipe_ci_script_master.sh (hunk 9 above)
Comment (craiglabenz) [reply tool/mediapipe_ci_script_master.sh:12]: It is!

File: tool/ci_script_shared.sh (hunk 6 above)
Comment (craiglabenz) [reply tool/ci_script_shared.sh]: finally removed!

File: packages/mediapipe-task-text/test/text_classifier_result_test.dart (hunk 5 above)
Comment (craiglabenz) [reply packages/mediapipe-task-text/test/text_classifier_result_test.dart]: restored

File: packages/mediapipe-task-text/test/text_classifier_result_test.dart (hunk 4 above)
Comment (craiglabenz) [reply packages/mediapipe-task-text/test/text_classifier_result_test.dart]: it is restored! (and I have no recollection of why it was ever commented out - possibly only to reduce noise from around the failing test I was debugging 🤷 )

File: packages/mediapipe-task-text/test/text_classifier_executor_test.dart [hunk 11]
```diff
@@ -1,10 +1,30 @@
+@Tags(['native-assets'])
+
 import 'dart:io' as io;
 import 'package:flutter_test/flutter_test.dart';
 import 'package:mediapipe_text/mediapipe_text.dart';
 import 'package:path/path.dart' as path;
 import 'package:mediapipe_core/mediapipe_core.dart';
 import 'package:mediapipe_text/src/tasks/text_classification/text_classification_executor.dart';
 
+void debugFileSystem() {
```
[the remark this answers is not in the document]
Comment (craiglabenz) [reply packages/mediapipe-task-text/test/text_classifier_executor_test.dart]: Yep, and that nonsense is gone now

**Pull request merged**